sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -933,3 +933,58 @@ case class JsonObjectKeys(child: Expression) extends UnaryExpression with Codege
     new GenericArrayData(arrayBufferOfKeys.toArray)
   }
 }
+
+/**
+ * A function which checks whether the given JSON string is valid JSON string or not.
+ */
+@ExpressionDescription(
+  usage = """
+    _FUNC_(json_string) - Returns `true` if `json_string` is valid JSON, `false` if `json_string` is
+      invalid. `NULL` is returned in case of `NULL`.
+  """,
+  examples = """
+    Examples:
+      > Select _FUNC_(null);
+        NULL
+      > Select _FUNC_('{"key" : "valid JSON"}');
+        true
+      > Select _FUNC_('{"invalid JSON"}');
```

Comment:
Could you add nested cases in the examples?

Reply:
Yes. Sure.